### PR TITLE
Fix lazy compilation of standard lib in `-resource-dir`

### PR DESCRIPTION
**Pull request by redsun82**

### CODE REVIEW

File: swift_tag.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+swift-5.7.1-RELEASE
```
[the remark this answers is not in the document]
Comment (redsun82) [reply swift_tag.txt:1]: whoops, a remnant of some experiment

File: patches/swift/03-reinstate-swiftinterface-compilation-for-resource-dir.patch [hunk 2]
```diff
@@ -0,0 +1,26 @@
+diff --git a/lib/Frontend/ModuleInterfaceLoader.cpp b/lib/Frontend/ModuleInterfaceLoader.cpp
```
[the remark this answers is not in the document]
Comment (redsun82) [reply patches/swift/03-reinstate-swiftinterface-compilation-for-resource-dir.patch:1]: I was sure I had written that

File: patch.sh [hunk 3]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+
+PATCH_DIR="$(cd "$(dirname "$0")/patches"; pwd)"
+
+for patch_subdir in "$PATCH_DIR"/*; do
+  (
+    repo="$(basename "$patch_subdir")"
+    cd "$repo" || exit 0
```
Comment (AlexDenisov) [inline patch.sh:8]: Should be followed by `cd ..` (or whatever the previous directory was)?

Comment (redsun82) [reply patch.sh:8]: no, because it's in a `(...)` subshell, so any change of directory is discarded on exit

**Pull request merged**